osu.Game/Overlays/Profile/UserGraph.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Cursor;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osuTK;
+
+namespace osu.Game.Overlays.Profile
+{
+    public abstract class UserGraph<TKey, TValue> : Container, IHasCustomTooltip
```

Comment:
xmldoc this class. xmldoc all abstract classes *completely*.

Reply:
There are still undocumented protected members (from the more important ones - `UserGraph.Data`, `UserGraph.GetDataPointHeight`). Rest is self-explanatory to a degree I suppose.